src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/FormPass.php

```diff
@@ -34,12 +34,17 @@ public function process(ContainerBuilder $container)
         $types = array();
 
         foreach ($container->findTaggedServiceIds('form.type') as $serviceId => $tag) {
-            $alias = isset($tag[0]['alias'])
-                ? $tag[0]['alias']
-                : $serviceId;
+            // The following if-else block is deprecated and can be removed
+            // in Symfony 3.0
+            if (isset($tag[0]['alias'])) {
+                $types[$tag[0]['alias']] = $serviceId;
+            } else {
+                $types[$serviceId] = $serviceId;
```

Comment:
And it would also account for @DavidBadura use-case in https://github.com/symfony/symfony/pull/15079#issuecomment-114884704 since the different services would have a different ID.